c++/CMakeLists.txt

```diff
@@ -25,7 +25,6 @@ set(INSTALL_TARGETS_DEFAULT_ARGS
 
 # Options ======================================================================
 
-option(BUILD_TESTING "Build unit tests and enable CTest 'check' target." ON)
```

Comment:
Hmm, does this break folks who use the `capnproto/c++/CMakeLists.txt` as the top-level CMakeLists.txt? As I recall, that file predates the `capnproto/CMakeLists.txt`, which, I think, was only introduced because Conan doesn't support non-root top-level CMakeLists.txt files.

Or was the `c++/CMakeLists.txt` always broken because we don't call `enable_testing()`?